### PR TITLE
feat: render binding in capsule instead of drop down

### LINKED ISSUE #414: Display protocol on bindings in capsule instead of drop down
The idea is to improve the actual bindings UI to match the initial implementation do in the html-template

**Reasons**

- improve the user experience when reading bindings blocks
- gain one level on rendering (because the protocol part is attached to the bindings title) into the bindings parts

**Attachments**

Here is the previous rendering (available on html-template 0.20) on the left side and the actual rendering on the right side

image

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

Render binding information as single component: in capsule instead of drop down.

Example:

image

**Related issue(s)**
Resolves #414

### CODE REVIEW

File: library/src/components/Bindings.tsx [hunk 1]
```diff
@@ -10,13 +10,28 @@ interface Props {
 }
 
 export const Bindings: React.FunctionComponent<Props> = ({
-  name = 'Bindings',
+  name = 'Binding',
```
Comment (M3lkior) [inline library/src/components/Bindings.tsx:13]: What about using the `specific information ` suffix as text (validated with @derberg in the original implementation). This provides a more comprehensive label for non-tech which are reading the generated documentation ;

File: library/src/containers/Operations/Operation.tsx [hunk 2]
```diff
@@ -118,7 +118,7 @@ export const Operation: React.FunctionComponent<Props> = ({
 
         {channel.hasBindings() && (
           <div className="mt-2">
-            <Bindings name="Channel Bindings" bindings={channel.bindings()} />
+            <Bindings name="Channel Binding" bindings={channel.bindings()} />
```
Comment (M3lkior) [inline library/src/containers/Operations/Operation.tsx:121]:
```suggestion
            <Bindings name="Channel specific information" bindings={channel.bindings()} />
```

File: library/src/containers/Operations/Operation.tsx [hunk 3]
```diff
@@ -127,7 +127,7 @@ export const Operation: React.FunctionComponent<Props> = ({
         {operation.hasBindings() && (
           <div className="mt-2">
             <Bindings
-              name="Operation Bindings"
+              name="Operation Binding"
```
Comment (M3lkior) [inline library/src/containers/Operations/Operation.tsx:130]:
```suggestion
              name="Operation specific information"
```

File: library/src/components/Bindings.tsx (hunk 1 above)
Comment (magicmatatjahu) [reply library/src/components/Bindings.tsx:13]: I changed it and it's an example how it looks:

image

Comment (M3lkior) [reply library/src/components/Bindings.tsx]: very nice 👌

**Comment by derberg** [comment]
what the?

Screenshot 2021-08-23 at 15 18 37

File: .sonarcloud.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Disable cognitive complexity in files
+sonar.exclusions=library/src/components/Schema.tsx
```
Comment (derberg) [inline .sonarcloud.properties:2]: I think it excludes all not just complexity, please ignore complexity on a file-level maybe -> https://github.com/asyncapi/parser-js/blob/master/test/testsUtils.js#L23

**Comment by magicmatatjahu** [comment]
Probably due to Ludovic's review.

**Review by derberg — COMMENTED** [review]
one comment, also why package-lock files had to be updated?

can you also update description of the PR with latest screen shot?

File: .sonarcloud.properties (hunk 4 above)
Comment (magicmatatjahu) [reply .sonarcloud.properties]: I tried to use the `eslint` and it doesn't work. I know about that sonar exclude all rules, but I cannot find example how to exclude only particular rule, I also cannot find the title/id of the complexity rule...

Also related to the package-lock.json - lerna's `bootstrap` removes the redundant modules in the monorepo :) I will revert this.

File: library/src/components/Schema.tsx [hunk 5]
```diff
@@ -1,11 +1,13 @@
+/* eslint-disable sonarjs/cognitive-complexity */
```
Comment (derberg) [inline library/src/components/Schema.tsx:1]: I think it works only if you put it right before the affected function

Comment (magicmatatjahu) [reply library/src/components/Schema.tsx]: Eslint doesn't work in any form, because then we should have errors from `lint` stage, and we don't.

**Comment by M3lkior** [comment]
> @derberg Probably due to Ludovic's review.

Hum .. i just leave single comment ; no review comment

**Review by derberg — APPROVED** [review]
I'm wondering if we can have a better PR title, but of course I have no better suggestion out of my head 😆

**Pull request merged**